package com.spring.weather.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity 
@Table (name = "users")
@Getter 
@Setter 
@AllArgsConstructor 
@NoArgsConstructor 
@Builder 
public class User {

    @Id 
    @GeneratedValue (strategy = GenerationType.IDENTITY)   
    private Long id;

    @Column (name = "email", nullable = false, length = 255)
    private String email;

    @Column (name = "password_hash", nullable = false, length = 255)
    private String passwordHash;

    @Column (name = "version", nullable = false)
    private Integer version;
}